src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -51,10 +61,71 @@ public boolean retainAll(Collection<?> c) {
     }
 
     public boolean removeAll(Collection<?> c) {
-        return false;
+        Objects.requireNonNull(c);
+        boolean var2 = false;
+        Iterator var3;
+        if (this.size() > c.size()) {
+            for(var3 = c.iterator(); var3.hasNext(); var2 |= this.remove(var3.next())) {
+            }
+        } else {
+            var3 = this.iterator();
+
+            while(var3.hasNext()) {
+                if (c.contains(var3.next())) {
+                    var3.remove();
+                    var2 = true;
+                }
+            }
+        }
+
+        return var2;
     }
 
     public void clear() {
 
     }
+    public static <E> Range<E> of(E first, E second) {
+        if (first instanceof Integer) {
+            return rangeInt((Integer) first, (Integer) second);
+        }
+        return rangeFloat((Float) first, (Float) second);
+    }
+
+    public static <E> Range<E> of(Character first, Character second, Function function) {
```

Comment:
You have to create an general solution for of method

Reply:
> You have to create an general solution for of method
What do you mean?
Like generic or overloading or what?